src/worldstate/include/routines/Gate/PassGate.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Created By: Joel Ahn
+ * Created On: March 5th, 2018
+ * Description: World State FSM node. Checks to see if
+ *              robot still needs to pass gate.
+ */
+
+#ifndef PROJECT_PASSGATE_H
+#define PROJECT_PASSGATE_H
+
+#include "State.h"
+#include <constants.h>
+#include <gate_detect/gateDetectMsg.h>
+
+/*** Communicating Class {alignWithGate, locatingGate, passGate} ***/
+class PassGate : public State {
+  public:
+    PassGate(int argc, char** argv, std::string node_name)
+      : State(argc, argv, node_name) {}
+
+    void setupNodeSubscriptions(ros::NodeHandle nh) override;
+    void sleep() override;
+
+  private:
+    ros::Subscriber gate_detect_listener;
+    // ros::Subscriber imu_listener_;
```

Comment:
We should remove commented out code or if the intent is to uncomment this for use later, add a line above with a issue reference, ie:
```
// TODO: we'll use this to listen for the imu, see issue #123
// ros::Subscriber imu_listener_;
```

Reply:
Wrote in the TODO because we might want to construct more complex logic in the future